lib/fluent/plugin/out_s3.rb

```diff
@@ -489,7 +493,19 @@ def setup_credentials
         credentials_options[:policy] = c.policy if c.policy
         credentials_options[:duration_seconds] = c.duration_seconds if c.duration_seconds
         credentials_options[:external_id] = c.external_id if c.external_id
-        if c.sts_region
+        credentials_options[:sts_endpoint_url] = c.sts_endpoint_url if c.sts_endpoint_url
+        credentials_options[:sts_http_proxy] = c.sts_http_proxy if c.sts_http_proxy
+        if c.sts_http_proxy and c.sts_endpoint_url
```

Comment:
Please use `&&` instead of `and` because they have different precedence. It's confusing.

https://ruby-doc.org/core-2.6.2/doc/syntax/precedence_rdoc.html#top

Reply:
sounds good, I'll update to use `&&` instead.